lib/readline.js

```diff
@@ -94,10 +95,20 @@ function Interface(input, output, completer, terminal) {
     if (input.prompt !== undefined) {
       prompt = input.prompt;
     }
+    if (input.escapeCodeTimeout !== undefined) {
+      this.escapeCodeTimeout = input.escapeCodeTimeout;
+    }
     crlfDelay = input.crlfDelay;
     input = input.input;
   }
 
+  if (!Number.isFinite(this.escapeCodeTimeout)) {
```

Comment:
This should be moved into the added upper if statement, since it is only necessary to test to possible provided option. If non was passed through, we already know that the value is finite.